src/util/self_driving_recording_util.cpp

```diff
@@ -37,9 +38,15 @@ void SelfDrivingRecordingUtil::RecordQueryMetadata(
 
   if (!params_vec.empty()) {
     std::string query = SelfDrivingRecordingUtil::QUERY_TEXT_INSERT_STMT;
-    task_manager->AddTask(std::make_unique<task::TaskDML>(catalog::INVALID_DATABASE_OID, query,
-                                                          std::make_unique<optimizer::TrivialCostModel>(), false,
-                                                          std::move(params_vec), std::move(param_types)));
+    task_manager->AddTask(
+        task::TaskDML::Builder()
+            .SetDatabaseOid(task_manager->GetDatabaseOid())
+            .SetQueryText(std::move(query))
+            // TODO(WAN): #1595
```

Comment:
Is #1595 the correct issue number? It's already closed...

Reply:
Yes, at the time I was not sure if it should be disabled durability (disabled durability didn't work). Good catch, refactoring to push it down

Reply:
Refactored